### PR TITLE
add five javadocs

### PR DESCRIPTION
**Pull request by andreyliporage**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: ---

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 1]
```diff
@@ -208,6 +208,17 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
         return identity;
     }
 
+    /**
+     * Creates a session identity.
+     * <p>
+     * Checks if the user is non-null, add header, add extraCredentials and add groups to build a session identity.
+     *
+     * @param authenticatedIdentity
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:216]: missing description

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 2]
```diff
@@ -208,6 +208,17 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
...
+     * @param headers a MultivaluedMap string
+     * @param groupProvider a Interface
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:218]: incomplete description

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 3]
```diff
@@ -348,6 +359,15 @@ public boolean supportClientTransaction()
         return queryRequestMetadata;
     }
 
+    /**
+     * Extracts a header from a headers map in the for of List<String>. Breaks each value with a comma, removes spaces and omits empty string,
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:363]: fix ponctuation

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 4]
```diff
@@ -379,11 +406,29 @@ public boolean supportClientTransaction()
         return roles.build();
     }
 
+    /**
+     * Extracts a specific header {@code PRESTO_EXTRA_CREDENTIAL}
+     *
+     * @param headers a MultivaluedMap string
+     *
+     * @return parseProperty
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:414]: missing description

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 5]
```diff
@@ -510,6 +567,18 @@ private static WebApplicationException badRequest(String message)
                 .build());
     }
 
+    /**
+     * Returns the given string if is non-null and nonempty.
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:571]: typo (non-empty)

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 6]
```diff
@@ -208,6 +208,17 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
         return identity;
     }
 
+    /**
+     * Creates a session identity.
+     * <p>
+     * Checks if the user is non-null, add header, add extraCredentials and add groups to build a session identity.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:214]: Your description should be more like a text, and not just steps.
I would do this description like this:
"Create a Session Identity according to the defined parameters headers, credentials and groups."

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 7]
```diff
@@ -348,6 +359,15 @@ public boolean supportClientTransaction()
         return queryRequestMetadata;
     }
 
+    /**
+     * Extracts a header from a headers map in the for of List<String>. Breaks each value with a comma, removes spaces and omits empty string.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:363]: Where is the summary? We should have a summary before complete descriptions.
Also your line seems too long, break it please

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 8]
```diff
@@ -363,6 +383,13 @@ public boolean supportClientTransaction()
         return parseProperty(headers, PRESTO_SESSION);
     }
 
+    /**
+     * Analyzes whether the {@code PRESTO_ROLE} header meets the standards. If it does, this header will be extracted,
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:387]: Same comment, where's the summary?

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java (hunk 6 above)
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:214]: You should also use more the link tag on the descriptions to point to important classes like the Session Identity

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 9]
```diff
@@ -379,11 +406,29 @@ public boolean supportClientTransaction()
         return roles.build();
     }
 
+    /**
+     * Extracts a specific header PRESTO_EXTRA_CREDENTIAL
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:410]: "Extracts the specific header PRESTO_EXTRA_CREDENTIAL"

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 10]
```diff
@@ -379,11 +406,29 @@ public boolean supportClientTransaction()
...
+     *
+     * @param headers a MultivaluedMap string
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:412]: This description is incomplete. What this map represents? All PRresto headers?

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 11]
```diff
@@ -379,11 +406,29 @@ public boolean supportClientTransaction()
...
+     *
+     * @return a Map with the {@link PRESTO_EXTRA_CREDENTIAL} properties
+     */
     private static Map<String, String> parseExtraCredentials(MultivaluedMap<String, String> headers)
     {
         return parseProperty(headers, PRESTO_EXTRA_CREDENTIAL);
     }
 
+    /**
+     * Returns a map string with parsed properties.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:422]: "Map string" seems strange to me. A better description would be:
"Returns a map containing all Presto properties." (Check if this is really the purpose)

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 12]
```diff
@@ -400,12 +445,28 @@ public boolean supportClientTransaction()
         return properties;
     }
 
+    /**
+     * Extracts a header from a headers map, break each value with a comma, remove spaces and omit empty strings.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:449]: Where is the summary?

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 13]
```diff
@@ -379,11 +412,29 @@ public boolean supportClientTransaction()
...
     }
 
+    /**
+     * Returns a map containing all Presto properties.
+     * <p>
+     * Validates that each header presents a set of key/value and adds the valid values in a HashMap properties. For invalid values, an
+     * IllegalArgumentException is generated and handled.
+     *
+     * @param headers a MultivaluedMap string
+     * @param headerName a string with the header name.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:434]: Ommit "a string"

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 14]
```diff
@@ -379,11 +412,29 @@ public boolean supportClientTransaction()
...
+     *
+     * @return a HashMap with the properties.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:436]: a HashMap with the extracted header properties.

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 15]
```diff
@@ -400,12 +451,32 @@ public boolean supportClientTransaction()
...
 
+    /**
+     * Extracts the header PRESTO_CLIENT_TAGS
+     * <p>
+     * Extracts a header from a headers map, break each value with a comma, remove spaces and omit empty strings.
+     * Returns a Set<String> with the extracted {@code PRESTO_CLIENT_TAGS} header.
+     *
+     * @param headers a MultivaluedMap string
+     *
+     * @return a Set<String> with {@code PRESTO_CLIENT_TAGS} headers.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:462]: a set with the ... headers

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 16]
```diff
@@ -400,12 +451,32 @@ public boolean supportClientTransaction()
         return properties;
     }
 
+    /**
+     * Extracts the header PRESTO_CLIENT_TAGS
+     * <p>
+     * Extracts a header from a headers map, break each value with a comma, remove spaces and omit empty strings.
+     * Returns a Set<String> with the extracted {@code PRESTO_CLIENT_TAGS} header.
+     *
+     * @param headers a MultivaluedMap string
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:460]: Same comment

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 17]
```diff
@@ -348,6 +357,19 @@ public boolean supportClientTransaction()
         return queryRequestMetadata;
     }
 
+    /**
+     * Returns a list of the HTTP headers received on the request.
+     * <p>
+     * Extracts the HTTP headers from the received request and creates a list
+     * of these headers to be returned.
+     * Breaks each value with a comma, removes spaces and omits empty string.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:365]: Remember that we defined that these information we are going to remove right?

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 18]
```diff
@@ -363,6 +386,16 @@ public boolean supportClientTransaction()
         return parseProperty(headers, PRESTO_SESSION);
     }
 
+    /**
+     * Checks if the header matches a pattern.
+     * <p>
+     * Analyzes whether the PRESTO_ROLE header meets the standards. If it does, this header will be extracted,
+     * if it doesn't, a WebApplicationException will be thrown.
```
[the remark this answers is not in the document]
Comment (joaopedro-simbiose) [reply presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:392]: Remember that we were going to change the first phrase "Analyzes whether the PRESTO_ROLE header meets the standards" to "Checks if the received headers matches the defined headers pattern."

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 19]
```diff
@@ -400,12 +452,32 @@ public boolean supportClientTransaction()
         return properties;
     }
 
+    /**
+     * Extracts the header PRESTO_CLIENT_TAGS
+     * <p>
+     * Extracts a header from a headers map, break each value with a comma, remove spaces and omit empty strings.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:458]: Remember that we were going to remove these description that are not necessary?

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 20]
```diff
@@ -400,12 +452,32 @@ public boolean supportClientTransaction()
...
     private static Set<String> parseClientTags(MultivaluedMap<String, String> headers)
     {
         Splitter splitter = Splitter.on(',').trimResults().omitEmptyStrings();
         return ImmutableSet.copyOf(splitter.split(nullToEmpty(headers.getFirst(PRESTO_CLIENT_TAGS))));
     }
 
+    /**
+     * Extracts the header PRESTO_CLIENT_CAPABILITIES.
+     * <p>
+     * Extracts a header from a headers map, break each value with a comma, remove spaces and omit empty strings.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:474]: Remember that these info are not necessary?

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 21]
```diff
@@ -453,6 +525,18 @@ private static ResourceEstimates parseResourceEstimate(MultivaluedMap<String, St
         return Optional.ofNullable(queryRequestMetadata);
     }
 
+    /**
+     * Its a generic check whether an expression is true or false.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:529]: Assert if a given expression is true or false.

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 22]
```diff
@@ -453,6 +525,18 @@ private static ResourceEstimates parseResourceEstimate(MultivaluedMap<String, St
...
+     * <p>
+     * Receives a boolean expression, a message and multiple arguments as parameters.
+     * If the expression is false, the method will throw an WebApplicationException.
+     *
+     * @param expression any boolean expression
+     * @param format message that will appear if it is false
+     * @param args multiple arguments that can be inserted in the format
+     *
+     * @throws WebApplicationException will be thrown if the expression is false.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:538]: @throws WebApplicationException if the expression is false.

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 23]
```diff
@@ -453,6 +525,18 @@ private static ResourceEstimates parseResourceEstimate(MultivaluedMap<String, St
...
     }
 
+    /**
+     * Its a generic check whether an expression is true or false.
+     * <p>
+     * Receives a boolean expression, a message and multiple arguments as parameters.
+     * If the expression is false, the method will throw an WebApplicationException.
+     *
+     * @param expression any boolean expression
+     * @param format message that will appear if it is false
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:535]: message that will be returned if the expression is false

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 24]
```diff
@@ -453,6 +525,18 @@ private static ResourceEstimates parseResourceEstimate(MultivaluedMap<String, St
...
+     * @param args multiple arguments that can be inserted in the format
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:536]: multiple arguments that can be inserted in the response message

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 25]
```diff
@@ -510,6 +594,18 @@ private static WebApplicationException badRequest(String message)
                 .build());
     }
 
+    /**
+     * Returns the given string if is non-null and non-empty.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:598]: I think we need just the summary: Check and return a given string if it is non-null and non-empty.

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 26]
```diff
@@ -510,6 +594,18 @@ private static WebApplicationException badRequest(String message)
...
+     * <p>
+     * Checks if the parameter is non-null and non-empty as well as checks the amount of bits
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:600]: I think we don't need this description. We may remove it

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 27]
```diff
@@ -510,6 +594,18 @@ private static WebApplicationException badRequest(String message)
...
+     * to classify it according to the Unicode standard.
+     *
+     * @param value a string that represents PrestoHeaders.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:603]: "a given string value"

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 28]
```diff
@@ -510,6 +594,18 @@ private static WebApplicationException badRequest(String message)
...
+     *
+     * @return {@code string} itself if it is non-null and non-empty.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:605]: "the given string if it is non-null and non-empty"
